src/lib/shim/shim_sys.c

```diff
@@ -142,6 +154,24 @@ bool shim_sys_handle_syscall_locally(long syscall_num, long* rv, va_list args) {
         }
     }
 
+    int straceFd = shim_straceFd();
+
+    if (straceFd >= 0) {
+        // TODO: format the time
+        uint64_t emulated_time_ms = shim_sys_get_simtime_nanos();
+        pid_t tid = shimshmem_getThreadId(shim_threadSharedMem());
+
+        bool oldNativeSyscallFlag = shim_swapAllowNativeSyscalls(true);
+        char buf[100] = {0};
+        int len = snprintf(buf, sizeof(buf), "%018ld [tid %d] %s(...) = %ld\n", emulated_time_ms,
+                           tid, syscallName, *rv);
+        len = MIN(len, sizeof(buf));
+        if (write(straceFd, buf, len) < 0) {
```

Comment:
technically I think we ought to loop here and keep track of how much was actually written. A successful `write` call could write anything from 0 to `len` bytes. OTOH this probably won't happen in practice.

Reply:
Yeah it seems to work fine in the shim log which just does one `write` call, but agreed that it's best to write it properly here. Fixed.